src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.java

```diff
@@ -68,40 +72,51 @@ public void updateAttendance(final long id, final UpdateAttendanceDto newAttenda
         final var attendance = attendanceRepository.findById(id).orElseThrow(AttendanceNotFound::new);
 
         if (isAttendanceLocked(attendance.getPaid(), attendance.getEventDate())) {
+            log.info("Update attempted on locked attendance, attendance id {}", id);
             throw new AttendanceLocked();
         }
 
+        handleIEPWarningScenarios(attendance, newAttendanceDetails).ifPresent(attendance::setCaseNoteId);
+
+        attendance.setComments(newAttendanceDetails.getComments());
+        attendance.setAttended(newAttendanceDetails.getAttended());
+        attendance.setPaid(newAttendanceDetails.getPaid());
+        attendance.setAbsentReason(newAttendanceDetails.getAbsentReason());
+
+        attendanceRepository.save(attendance);
+
+        postNomisAttendance(attendance);
+    }
+
+    private Optional<Long> handleIEPWarningScenarios(final Attendance attendance, final UpdateAttendanceDto newAttendanceDetails) {
         final var shouldTriggerIEPWarning =
                 newAttendanceDetails.getAbsentReason() != null &&
                         AbsentReason.getIepTriggers().contains(newAttendanceDetails.getAbsentReason());
 
         final var shouldRevokePreviousIEPWarning = attendance.getCaseNoteId() != null && !shouldTriggerIEPWarning;
 
         if (shouldRevokePreviousIEPWarning) {
-            final var rescindedReason = "IEP rescinded: " + (newAttendanceDetails.getAttended() ?
-                    "attended" : newAttendanceDetails.getAbsentReason().toString());
+            final var rescindedReason = "IEP rescinded: " + (
+                    newAttendanceDetails.getAttended() ? "attended" :
+                            capitalize(lowerCase(join(splitByCharacterTypeCamelCase(newAttendanceDetails.getAbsentReason().toString()), ' ')))
+            );
```

Comment:
Could this be assigned to a variable like `formattedAbsentReason` or something like that so the logic isn't part of this if statement?